### DISCUSSION TITLE
Apollo useQuery with parameters from NextJs useRouter

### DISCUSSION THREAD

**Discussion by indiehjaerta** [vercel/next.js#17526] [Help]
Looking for answer and even possibly  an example how I could go about using the id from the useRouter from 'next/router' only when it is resolved.

```
const User = () => {
	const router = useRouter();

	const { loading, error, data } = useQuery(USER_QUERY, {
		variables: { slug: router.query.slug },
	});

	return <div>My render</div>;
};

```

in this case on the first render router.query.slug would be undefined meaning I would make a call to useQuery with a missing paramater resulting in a 400 call and I would like it to only run whenever router has resolved.

![image](https://user-images.githubusercontent.com/14825674/94822726-44c36080-0403-11eb-91aa-5a4281d7f042.png)

So if anyone have had encountered the same issue and resolved it, I would appriciate the help.

**Comment by jamesmosier** [comment]
It looks like `useQuery` has a [`skip` option](https://www.apollographql.com/docs/react/data/queries/#options). You could wait for `router` to have a value then change `skip` to `false` to trigger the query.

```js
const router = useRouter();
const slug = router.query.slug;

const [shouldSkip, setShouldSkip] = React.useState(true)
const { loading, error, data } = useQuery(USER_QUERY, {
  variables: { slug: router.query.slug },
  skip: shouldSkip
});

React.useEffect(() => {
  if (slug) {
    setShouldSkip(false)
  }
}, [slug]);
```

**Reply by hermanskurichin** [reply]
router also exposes `isReady` prop, we probably can toggle skip based on that.

```Javascript
const router = useRouter();
const slug = router.query.slug;

const [shouldSkip, setShouldSkip] = React.useState(true)
const { loading, error, data } = useQuery(USER_QUERY, {
  variables: { slug: router.query.slug },
  skip: !router.isReady
});
```

**Reply by alekswithakayy** [reply]
This solution works fine during dev but as soon as I try to build my site I get an error: 

```
Error occurred prerendering page "/tokens/[fingerprint]". Read more: https://nextjs.org/docs/messages/prerender-error
undefined
undefined
```

Taking away the skip parameter resolves the build error. Any idea how to resolve this?

More info: https://stackoverflow.com/questions/73780579/next-js-errors-when-building-dynamically-routed-pages-that-use-apollo-usequery-a
